far/vmenu.cpp

```diff
@@ -1434,6 +1501,17 @@ bool VMenu::ProcessKey(const Manager::Key& Key)
 			}
 		}
 	};
+
+	const auto HScrollSmartPos = [=]()
```

Comment:
- I think I understand the intention of `[=]` (*"we only need to capture LocalKey, which is an int or something, doing so by value means fewer indirections and better inlining"*), but by value capture can have funny consequences if we extend this lambda further one day, capture more and forget to update it accordingly.
In general, I'd advise against this syntax entirely, unless it is absolutely necessary to capture *everything* by value (e.g. this is some async callback that outlives the outer context).
Something like `[LocalKey]` should be more fool- and future-proof, but I also don't see why do we need to micro-optimize this piece at all instead of just using `[&]`: it will likely be fully inlined in either case, and it's a keyboard handler, it won't be called more than 30 or so times per second anyway. 
- `()` probably could be omitted.

Same for the other lambdas in this file.

Reply:
Well, my philosophy is to pass everything by value (to follow the general language approach) and delete copy / move constructor whenever it makes sense. This way I can use `[=]` everywhere and still stay reasonably future-proof.

Anyway, I replaced `[=]` with `[&]` everywhere and removed `()` whenever possible. Thanks for point this out.